b_pull_Landsat_SRST_poi/py/runGEEperTile.py

```diff
@@ -0,0 +1,342 @@
+  .filterMetadata("PR", "equals", tiles))
+
+#grab images and apply scaling factors
+l7 = (ee.ImageCollection("LANDSAT/LE07/C02/T1_L2")
+    .map(apply_scale_factors)
+    .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end))
+l5 = (ee.ImageCollection("LANDSAT/LT05/C02/T1_L2")
+    .map(apply_scale_factors)
+    .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end))
+l4 = (ee.ImageCollection("LANDSAT/LT04/C02/T1_L2")
+    .map(apply_scale_factors)
+    .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end))
+    
+# merge collections by image processing groups
+ls457 = (ee.ImageCollection(l4.merge(l5).merge(l7))
+    .filterBounds(wrs))  
+    
+# existing band names
+bn457 = (["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B7", 
+  "QA_PIXEL", "SR_CLOUD_QA", "QA_RADSAT", "ST_B6", 
+  "ST_QA", "ST_CDIST", "ST_ATRAN", "ST_DRAD", "ST_EMIS",
+  "ST_EMSD", "ST_TRAD", "ST_URAD"])
+  
+# new band names
+bns457 = (["Blue", "Green", "Red", "Nir", "Swir1", "Swir2", 
+  "pixel_qa", "cloud_qa", "radsat_qa", "SurfaceTemp", 
+  "temp_qa", "ST_CDIST", "ST_ATRAN", "ST_DRAD", "ST_EMIS",
+  "ST_EMSD", "ST_TRAD", "ST_URAD"])
+  
+# rename bands  
+ls457 = ls457.select(bn457, bns457)
+
+
+#grab images and apply scaling factors
+l8 = (ee.ImageCollection("LANDSAT/LC08/C02/T1_L2")
+    .map(apply_scale_factors)
+    .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end))
+l9 = (ee.ImageCollection("LANDSAT/LC09/C02/T1_L2")
+    .map(apply_scale_factors)
+    .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
+    .filterDate(yml_start, yml_end))
+
+# merge collections by image processing groups
+ls89 = ee.ImageCollection(l8.merge(l9)).filterBounds(wrs)  
+    
+# existing band names
+bn89 = (["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", "SR_B7", 
+  "QA_PIXEL", "SR_QA_AEROSOL", "QA_RADSAT", "ST_B10", 
+  "ST_QA", "ST_CDIST", "ST_ATRAN", "ST_DRAD", "ST_EMIS",
+  "ST_EMSD", "ST_TRAD", "ST_URAD"])
+  
+# new band names
+bns89 = (["Aerosol","Blue", "Green", "Red", "Nir", "Swir1", "Swir2",
+  "pixel_qa", "aerosol_qa", "radsat_qa", "SurfaceTemp", 
+  "temp_qa", "ST_CDIST", "ST_ATRAN", "ST_DRAD", "ST_EMIS",
+  "ST_EMSD", "ST_TRAD", "ST_URAD"])
+ 
+# rename bands  
+ls89 = ls89.select(bn89, bns89)
+
+
+##########################################
+##---- LANDSAT 457 SITE ACQUISITION ----##
+##########################################
+
+## run the pull for LS457
+if "site" in extent:
+  
+  geo = wrs.geometry()
+  
+  ## get locs feature and buffer ##
+  feat = (locs_feature
+    .filterBounds(geo)
+    .map(dp_buff))
+      
+  ## process 457 stack
+  #snip the ls data by the geometry of the location points    
+  locs_stack_ls457 = ls457.filterBounds(feat.geometry()) 
```

Comment:
not a major point. but you might get faster speed query the satellite image by its path and row number since this is running by tiles.